src/PIL/DdsImagePlugin.py

```diff
@@ -148,15 +148,17 @@ def _open(self):
             self.tile = [("raw", (0, 0) + self.size, 0, (self.mode, 0, 1))]
         elif pfflags & DDPF_RGB:
             # Texture contains uncompressed RGB data
-            masks = {mask: ["R", "G", "B", "A"][i] for i, mask in enumerate(masks)}
-            rawmode = ""
-            if pfflags & DDPF_ALPHAPIXELS:
-                rawmode += masks[0xFF000000]
-            else:
+            if not pfflags & DDPF_ALPHAPIXELS:
                 self._mode = "RGB"
-            rawmode += masks[0xFF0000] + masks[0xFF00] + masks[0xFF]
+            if masks[:3] == (31744, 992, 31) and self.mode == "RGB":
```

Comment:
The magic numbers here could use some explanation if you ask me.

Reply:
I know what you're said sounds like a perfectly straightforward request, but I can't think of a short way to explain it that isn't just explaining what masks are, or what BGR;15 is.

There are 4 bit masks - https://learn.microsoft.com/en-us/windows/win32/direct3ddds/dds-pixelformat - R, G, B and A.

Each mask filters the pixel values to transform them into the value for each channel.

These masks are filtering 5 bits for R, 5 bits for G and 5 bits for A - BGR;15.
```
31744 = 0b0111110000000000
  992 = 0b0000001111100000
   31 = 0b0000000000011111
```

Reply:
That looks like great material for a code comment right there :)

You could also replace the decimal constants with those binary constants – then it's immediately more clear for a reader.

Reply:
Clearer, yes, it just feels verbose... I've pushed a commit.